tensorflow_io/core/plugins/s3/s3_filesystem.cc

```diff
@@ -125,36 +157,10 @@ static Aws::Client::ClientConfiguration& GetDefaultClientConfig() {
   if (!init) {
     const char* endpoint = getenv("S3_ENDPOINT");
     if (endpoint) cfg.endpointOverride = Aws::String(endpoint);
-    const char* region = getenv("AWS_REGION");
     // TODO (yongtang): `S3_REGION` should be deprecated after 2.0.
```

Comment:
@yongtang 
Maybe we should mark it as deprecated in `2.5rc` and remove it in `2.5` ?

Reply:
@vnvo2409 The comment was copied from tensorflow's source code. I think we can just remove it in the code, and add a note to the documentation. That should be enough (as it was supposed to be deprecated from 2.0 and we are in 2.4/2.5 already now).

Reply:
Done